### PR TITLE
Documentation improvement

### PR DESCRIPTION
**Pull request by bitbeast18**
Signed-off-by: Harshal Nandigramwar <harshal.11711777@lpu.in>

#### What it does
1. Fixes typo in the node version mentioned in the Building on Windows section of Developing.md guide.
2. Adds a reminder in the troubleshooting section in case somebody missed to read the docs.

#### How to test
This is a documentation enhancement and doesn't affect any code.

#### Review checklist

- [-] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by bitbeast18** [comment]
> @bitbeast18 thank you for taking care of the issue, however please revert any unrelated changes (such as formatting).

Thanks for reviewing @vince-fugnitto. I fixed the bad formatting.

**Comment by vince-fugnitto** [comment]
@bitbeast18 please be sure to squash your commits, and `sign-off` before the pull-request can be accepted (at the moment the `eclipsefdn/eca` check is failing due to a missing `sign-off`).

**Comment by bitbeast18** [comment]
> @bitbeast18 please be sure to squash your commits, and `sign-off` before the pull-request can be accepted (at the moment the `eclipsefdn/eca` check is failing due to a missing `sign-off`).

@vince-fugnitto Squashed and signed. This is my first ever PR. Thanks for directions.

**Comment by bitbeast18** [comment]
@vince-fugnitto 
Is there some problem with the CI build system? There were no code changes and the last commit was passing the build but still, this one fails randomly.

Here is the `stderr` of 3 tries with some formatting https://gist.github.com/bitbeast18/2868be8ea1508fda1eadc7a2a279823c

For all the three tries there were warnings like these
```
root WARN Commiting pending changes before change accessor leaves due to read access.
root WARN Commiting pending changes before change accessor leaves due to read access.
```

but according to #7481 this was removed. Btw, the test was failing for `yarn run test:browser` 

And then there was this unhandled error:

```
Error: Canceled: Canceled
    at Object.canceled (http://127.0.0.1:3000/vs/editor/editor.main.js:1618:21)
    at Delayer.cancel (http://127.0.0.1:3000/vs/editor/editor.main.js:9087:42)
    at http://127.0.0.1:3000/vs/editor/editor.main.js:116836:103
    at Object.dispose (http://127.0.0.1:3000/vs/editor/editor.main.js:5996:17)
    at http://127.0.0.1:3000/vs/editor/editor.main.js:6024:67
    at Set.forEach (<anonymous>)
    at DisposableStore.clear (http://127.0.0.1:3000/vs/editor/editor.main.js:6024:29)
    at DisposableStore.dispose (http://127.0.0.1:3000/vs/editor/editor.main.js:6018:18)
    at FindWidget.Disposable.dispose (http://127.0.0.1:3000/vs/editor/editor.main.js:6053:25)
    at http://127.0.0.1:3000/vs/editor/editor.main.js:6024:67
```
and a few `Error: TypeError: Cannot read property 'addWidget' of null`.

I don't exactly understand the problem.

File: doc/Developing.md [hunk 1]
```diff
@@ -338,6 +338,8 @@ Using Git Bash as administrator:
 
 If you do not have Git Bash installed on your system, [get one](https://gitforwindows.org/), or use `scoop`: `scoop install git`.
 
+If you want to run the electron based example then make sure to re-build the native packages by running `yarn run rebuild:electron` from the root directory. Note that if you wish to switch back to the browser example then you have to roll back these changes by running `yarn run rebuild:browser`.
```
Comment (akosyakov) [inline doc/Developing.md:341]: I don't think we need to repeat ourself. There is TOC and there is a section explaining how to run Electron which says one must run `yarn run rebuild:electron` before. If someone skips reading docs and goes to troubleshooting it is strange. Maybe just add a reminder instead that read docs first. The same in linux section.

Comment (bitbeast18) [reply doc/Developing.md:341]: Ok @akosyakov . So, shall I replace this with the following line

```
For running the electron example follow the (Running electron-based example)[https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#run-the-browser-based-example-application] section above.
```
or shall I completely remove it?

Also, should I keep the error message in the troubleshooting example below? and just add `First make sure that you follow the steps given in the docs correctly.`

Comment (akosyakov) [reply doc/Developing.md:341]: I would prefer to remove it completely and add a notice at the start of `Troubleshooting`

> First make sure that you follow the steps given in the docs correctly.

You can use `>` before to highlight it.

Comment (bitbeast18) [reply doc/Developing.md]: Done!

**Pull request merged**